instance/tests/models/test_log_entry.py

```diff
@@ -133,6 +133,36 @@ def test_log_publish(self, mock_publish_data):
             'server_id': self.server.pk,
         })
 
+    def test_log_delete(self):
+        """
+        Check `log_entries` output for combination of instance & server logs
+        """
+        server1 = self.server
+        server2 = OpenStackServerFactory(instance=self.instance, openstack_id='vm2_id')
+
+        self.instance.logger.info('Line #1, on instance')
+        server1.logger.info('Line #2, on server 1')
+        server2.logger.info('Line #3, on server 2')
+
+        self.assertEqual(LogEntry.objects.count(), 3)
+        # Delete server 1:
+        server1.delete()
+        # Now its log entry should be deleted:
+        self.assertEqual(LogEntry.objects.count(), 2)
+        entries = LogEntry.objects.order_by('pk').all().values_list('text', flat=True)
+        self.assertIn('Line #1, on instance', entries[0])
+        self.assertIn('Line #3, on server 2', entries[1])
```

Comment:
@bradenmacdonald Nit: Maybe add another check asserting that "Line #2, on server 1" is missing from `entries[1]`.